src/main/java/sirius/biz/storage/layer3/uplink/ftp/FTPUplink.java

```diff
@@ -185,25 +204,35 @@ private MutableVirtualFile wrap(VirtualFile parent, FTPFile file, String filenam
             return Optional.of(result);
         }
 
-        try (UplinkConnector<FTPClient> connector = connectorPool.obtain(ftpConfig)) {
-            FTPFile[] ftpFiles = list(connector.connector(),
-                                      file.parent().as(RemotePath.class),
-                                      ftpFile -> Strings.areEqual(ftpFile.getName(), file.name()));
-            if (ftpFiles.length == 1) {
-                file.attach(ftpFiles[0]);
-                return Optional.of(ftpFiles[0]);
-            } else {
-                return Optional.empty();
+        for (Attempt attempt : Attempt.values()) {
```

Comment:
can we put that for loop & error handling in an extra method with a runnable or consumer as parameter? 
This looks like copy & paste on 3 different places. Maybe:

`attemptConnection(Consumer<UplinkConnector>)`

Reply:
doesn't work within a lambda we need an inner return point